src/traces/isosurface/attributes.js

```diff
@@ -0,0 +1,184 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var colorscaleAttrs = require('../../components/colorscale/attributes');
+var colorbarAttrs = require('../../components/colorbar/attributes');
+var surfaceAtts = require('../surface/attributes');
+var baseAttrs = require('../../plots/attributes');
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+module.exports = extendFlat({
+    x: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: [
+            'Sets the X coordinates of the vertices on X axis.'
+        ].join(' ')
+    },
+    y: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: [
+            'Sets the Y coordinates of the vertices on Y axis.'
+        ].join(' ')
+    },
+    z: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: [
+            'Sets the Z coordinates of the vertices on Z axis.'
+        ].join(' ')
+    },
+    volume: {
```

Comment:
Why did you change `value` -> `volume`?

Reply:
We could use `value` there. But I liked your comment here: 
https://github.com/plotly/plotly.js/pull/2752#discussion_r201448179

Reply:
Ah nice find. Thanks! Let's keep `volume` for now. I might change my mind again, and we could use other opinions.

Reply:
hmm, I don't necessarily read that [matlab doc page](https://www.mathworks.com/help/matlab/ref/isosurface.html) as calling it "volume", they say "volume data V" but I read that as "V is a 3D array" rather than "V is called volume". I think I prefer `value`; `volume` implies that it's an array of volumes, but really it's an array of the value of some field/function dispersed over a volume.

Reply:
OK. The `value` or `values` could be applied in the next revision. The previous isosurface & volume traces used both two variations! 
@alexcjohnson please let me know which one to pick?

Reply:
The `value` is applied instead of `volume` and the mocks are codepen demos are updated in case @empet wanted to play with this trace...